cmd/api/main.go

```diff
@@ -7,15 +7,22 @@ import (
 	"net/http"
 
 	"github.com/flared/lokify/pkg/api"
+	"github.com/flared/lokify/pkg/loki"
 )
 
 var (
-	port = flag.String("port", "8080", "Define on which port the server will run. Default: 8080")
+	port        = flag.String("port", "8080", "Define on which port the server will run. Default: 8080")
+	lokiBaseUrl = flag.String("loki-base-url", "http://loki:3100", "Define loki base url. Defautl: http://loki:3100")
 )
 
 func main() {
 	flag.Parse()
 
+	lokiClient := loki.NewClient(&http.Client{}, *lokiBaseUrl)
+	lokifyApi := api.New(lokiClient)
+	router := lokifyApi.NewRouter()
```

Comment:
NewRouter should accept the api, not the reverse.

```go
	lokiClient := loki.NewClient(&http.Client{}, *lokiBaseUrl)
	lokifyApi := api.NewRouter(lokiClient)
```

It seems that you have one extra unnecessary level of abstraction.